src/java.xml/share/classes/com/sun/org/apache/xerces/internal/impl/XMLEntityManager.java

```diff
@@ -1209,11 +1209,12 @@ public XMLInputSource resolveEntity(XMLResourceIdentifier resourceIdentifier) th
 
         // Step 1: custom Entity resolver
         XMLInputSource xmlInputSource = null;
-
+        boolean resolveByResolver = false;
         if (fEntityResolver != null) {
             resourceIdentifier.setBaseSystemId(baseSystemId);
             resourceIdentifier.setExpandedSystemId(expandedSystemId);
             xmlInputSource = fEntityResolver.resolveEntity(resourceIdentifier);
+            resolveByResolver = (xmlInputSource != null) ? true : false;
```

Comment:
The "? true : false" isn't needed; the boolean result is sufficient.

Reply:
Thanks Roger. Fixed.